### PR TITLE
Consume dotnet-deb-tool in cli

### PR DESCRIPTION
**Pull request by brthor**
Last PR got messed up with the addition of new build definitions to Jenkins.

This PR removes `DebianPackageTool` from the cli repository and consumes it from the dotnet-cli feed where it is published by the `core-setup` build. There were a few changes to the DebPackageCreator to support the SDK deb file that I will backport to `core-setup` after this PR goes in.

The `DebianPackageTool` has been renamed to `dotnet-deb-tool` and is shipped as a CLI tool, which means it is referenced in the `tools` section of a project.json.

PTAL @eerhardt @livarcocc @piotrpMSFT

### CODE REVIEW

**Comment by eerhardt** [comment]
> // Copyright (c) .NET Foundation and contributors. All rights reserved.

Is there any way to call out the "few changes" to this file?  I can review the whole thing again, if you want.

---
Refers to: build_projects/shared-build-targets-utils/Utils/DebPackageCreator.cs:1 in dfa1178. [](commit_id = dfa1178df4501292ea2cdb75e541938ddfb67eaa, deletion_comment = False)

File: build_projects/dotnet-cli-build/DebTargets.cs [hunk 1]
```diff
@@ -28,33 +28,36 @@ public static BuildTargetResult GenerateSdkDeb(BuildTargetContext c)
...
-
-            if (Directory.Exists(objRoot))
+            var debianConfigVariables = new Dictionary<string, string>()
             {
-                Directory.Delete(objRoot, true);
-            }
+                { "SHARED_FRAMEWORK_DEBIAN_PACKAGE_NAME", sharedFxDebianPackageName },
+                { "SHARED_FRAMEWORK_NUGET_NAME", Monikers.SharedFrameworkName },
+                { "SHARED_FRAMEWORK_NUGET_VERSION",  CliDependencyVersions.SharedFrameworkVersion },
+                { "SHARED_FRAMEWORK_BRAND_NAME", Monikers.SharedFxBrandName },
```
Comment (eerhardt) [inline build_projects/dotnet-cli-build/DebTargets.cs]: Are all these "SHARED_FRAMEWORK" variables being used in the config.json file?  I only see `SHARED_FRAMEWORK_DEBIAN_PACKAGE_NAME` used.

**Comment by eerhardt** [comment]
>         "clang-3.5" : {},

Is the SDK directly dependent on 'clang'?

---
Refers to: packaging/deb/dotnet-debian_config.json:32 in dfa1178. [](commit_id = dfa1178df4501292ea2cdb75e541938ddfb67eaa, deletion_comment = False)

**Comment by eerhardt** [comment]
> // Copyright (c) .NET Foundation and contributors. All rights reserved.

I was able to do the diff offline. Looks like "versioning the manpages" and adding "debian" files were added.

The only comment I have on the new functionality is: "Shouldn't the `dotnet-deb-tool` do the work of versioning the manpages?"  I'm confused between what the difference is between the roles of the tool and of this DebPackageCreator.

---
In reply to: [222997618](https://github.com/dotnet/cli/pull/3312#issuecomment-222997618) [](ancestors = 222997618)

---
Refers to: build_projects/shared-build-targets-utils/Utils/DebPackageCreator.cs:1 in dfa1178. [](commit_id = dfa1178df4501292ea2cdb75e541938ddfb67eaa, deletion_comment = False)

**Comment by eerhardt** [comment]
:shipit:

**Comment by brthor** [comment]
@dotnet-bot test Ubuntu x64 Release Build please

**Comment by brthor** [comment]
@dotnet-bot test RHEL7.2 x64 Release Build please

**Comment by brthor** [comment]
>         "clang-3.5" : {},

No will remove this.

---
In reply to: [222999880](https://github.com/dotnet/cli/pull/3312#issuecomment-222999880) [](ancestors = 222999880)

---
Refers to: packaging/debian/dotnet-debian_config.json:32 in dfa1178. [](commit_id = dfa1178df4501292ea2cdb75e541938ddfb67eaa, deletion_comment = False)

**Comment by brthor** [comment]
@dotnet-bot test Ubuntu x64 Release Build please

**Comment by brthor** [comment]
@dotnet-bot test Ubuntu x64 Release Build please

**Comment by brthor** [comment]
@dotnet-bot test Ubuntu x64 Release Build

**Comment by brthor** [comment]
@dotnet-bot test Ubuntu x64 Release Build please

**Comment by brthor** [comment]
@dotnet-bot test Ubuntu x64 Release Build please

**Comment by brthor** [comment]
@dotnet-bot test Ubuntu x64 Release Build please

File: build_projects/dotnet-cli-build/DebTargets.cs (hunk 1 above)
Comment (brthor) [inline build_projects/dotnet-cli-build/DebTargets.cs]: Hmm I copied the substitutions from the the previous script, I will remove the unnecessary ones

**Comment by brthor** [comment]
> Is the SDK directly dependent on 'clang'?

Actually going to fix this in a separate PR as it needs to go into preview2 as well.

**Comment by eerhardt** [comment]
:shipit:

**Pull request merged**